### ISSUE TITLE
Units healing twice during a turn

### ISSUE THREAD

**Issue by Edaka** [LoneGazebo/Community-Patch-DLL#495]
I've noticed that sometimes a unit will be healed twice during one turn: once at the beginning of a human's turn and once at the beginning of an AI's turn. This happens when a unit is fortified in friendly territory and is being bombarded by a hostile ranged unit, after which it's being told to fortify again. I'm guessing it has to do something with units being automatically woken up each time they're being shot at.
Unfortunately, I don't have any relevant saves, but will try to look out for the issue next time it happens for me.

**Comment by Radost** [comment]
Are you sure it's healed twice and not just numbers showing healing from the previous turns?
I also thought it was healing several times, but upon closer inspection of turn-by-turn I didn't get it.

**Comment by Edaka** [comment]
I've just encountered the issue again, but this time I have a save that, hopefully, might be of use. I understand that providing logs would be much more useful and I'm sorry I don't know how to.
The save: https://www.dropbox.com/s/jpf5zg5xc763ixt/Catherine_0025%20BC-3375.Civ5Save?dl=0
I am absolutely sure that my warrior here is being healed twice per turn, as described in the OP, while being shot at by a barbarian archer. Usually all human's units heal at the beginning of an AI's turn, but this one here is healed at the beginning of my turn as well. 
Here's how it went for me: the warrior had 51 hp, I set him to fortify and press next turn - it's instantly healed by 15 (up to 66) hp. The AI civs take their turns, then the barbs - the archer shoots and brings my warrior's HP down by 12 (to 54). As the barbs finish their turn, my turn commences and at the same time the warrior is healed by 15 (up to 69) again.
